libvast/src/system/archive.cpp

```diff
@@ -72,13 +72,18 @@ archive(archive_type::stateful_pointer<archive_state> self,
         return make_error(ec::no_error);
       }
       std::vector<event> result;
-      auto slices = self->state.store->get(xs);
-      if (!slices)
-        VAST_DEBUG(self, "failed to lookup IDs in store:",
-                   self->system().render(slices.error()));
-      else
-        for (auto& slice : *slices)
-          to_events(result, *slice, xs);
+      auto session = self->state.store->extract(xs);
+      while (true) {
+        auto slice = session->next();
+        if (!slice) {
+          // Either we are done ...
+          if (!slice.error())
+            break;
+          // ... or an error occured.
```

Comment:
To save some vertical whitespace, you could place the comments after the respective lines.